scripts/genVis.py

```diff
@@ -1,58 +1,140 @@
+__author__ = 'Jon Mallen'
+
 from sklearn.manifold import TSNE
-#from MulticoreTSNE import MulticoreTSNE as TSNE
 from sklearn.decomposition import PCA
-import numpy as np
+import umap
 import matplotlib.pyplot as plt
-import matplotlib.cm as cm
 import pandas as pd
-from optparse import OptionParser 
-parser = OptionParser()
+import argparse
```

Comment:
Great question! I too asked the same thing before making the switch from `OptionParser` to `argparse`. The reason is flexibility. A careful user specifies the dimension they wish for each dimensional reduction as a value after each argument. However, sometimes the user might forget to specify a dimension value, and in that case, I believe it should just default to 2 instead of failing altogether. Therefore, the argument should be able to take in 0 or 1 values (The way it is set up, theoretically it could take in more than 1 value but only the first value would matter). Take t-SNE, for example. I can argue `-t` and the script would return a 2D t-SNE plot.  Or, I can argue `-t 2` or `-t 3` to return 2D or 3D t-SNE, respectively. It is `argparse` that allows for this flexibility, while `OptionParser` from `optparse` does not have this same functionality.